zap/src/test/java/org/zaproxy/zap/extension/anticsrf/ExtensionAntiCSRFUnitTest.java

```diff
@@ -351,6 +351,20 @@ void shouldGenerateFormWithEscapeHtmlChars() throws Exception {
             assertThat(form, is(equalTo(expectedForm(uri, params))));
         }
 
+        @Test
+        void shouldGenerateFormWithCustomActionUrl() throws Exception {
+            // Given
+            TreeSet<HtmlParameter> params =
+                    params(param("Name1", "Value1"), param("Name2", "Value2"));
+            given(message.getFormParams()).willReturn(params);
+            String actionUri = "http://example.com/form";
+            given(httpRequestHeader.getURI()).willReturn(uri(actionUri));
```

Comment:
Should be a different URI, otherwise we can't be sure which one is actually used.

Reply:
I mean the one returned by the request header needs to be different than the one in `actionUri`.